mips/uart_cbus.c

```diff
@@ -29,7 +31,7 @@ void uart_init() {
   LCR = LCR_8BITS; /* 8-bit data, no parity */
 }
 
-int uart_putchar(int c) {
+static void cbus_uart_putc(console_t *dev __unused, int c) {
```

Comment:
The driver will allocate a cyclic buffer (80 characters?) from generic memory pool available to all device drivers. `cn_putc` will put a character into the buffer, and then check if THRE (transmitter hold register empty) bit is set. If so it will initiate transfer that will be continued by interrupt handler routine if the buffer contains more than one character. If buffer is full the thread will be suspended. It will be woken up by interrupt handler routine when there's some space in the buffer to put data in.